app/models/rent.rb

```diff
@@ -3,4 +3,23 @@ class Rent < ApplicationRecord
   belongs_to :book, optional: true
 
   validates :start_date, :end_date, presence: true
+  validate :check_dates
+
+  def valid_dates?
+    in_range? && today_valid?
+  end
+
+  def in_range?
+    start_date < end_date
+  end
+
+  def today_valid?
+    start_date < Time.zone.today && Time.zone.today < end_date
```

Comment:
we didn't specified that a rent must have this validation

you have to validate that there's no other rent with the same book and user that intersect the dates with the one you are creating/updating

Reply:
<img width="577" alt="screen shot 2018-10-22 at 12 59 07 pm" src="https://user-images.githubusercontent.com/43754771/47309540-50dcfb80-d5fa-11e8-94a5-230f38716376.png">
look this